demos/mantle_convection/adaptive_base_case/adaptive_base_case.py

```diff
@@ -0,0 +1,362 @@
+# Mantle convection base case using adaptive meshing
+# ==================================================
+# This tutorial demonstrates the adaptive meshing capabality available in
+# G-ADOPT, which dynamically adapts the mesh to focus resolution where needed.
+# As demonstrated in [Davies et al.
+# (2011)](https://doi.org/10.1029/2011GC003551) this may significantly reduce
+# the computational requirements of mantle convection models, in particular
+# when using anisotropic metric-based adaptivity which can very efficiently
+# resolve the anisotropic features in geodynamical flows.
+#
+# Additional installation instructions
+# ------------------------------------
+# This functionality is available in G-ADOPT via the [mmg remeshing
+# library](https://www.mmgtools.org/) and therefore requires a few extra flags
+# in the configuration of the PETSc library used in the firedrake installation
+# as explained
+# [here](https://github.com/mesh-adaptation/docs/wiki/Installation-Instructions).
+# Additionally, for the assembly of the metric field, which allows us to
+# specify exactly where mesh resolution is needed, we use the
+# [animate](https://mesh-adaptation.github.io/docs/animate/index.html) python
+# package that can be pip installed from
+# https://github.com/mesh-adaptation/animate
+
+from gadopt import *
+from animate import RiemannianMetric, adapt
+
+# Set up the initial mesh and timestepping options
+
+nx, ny = 10, 10
+mesh = UnitSquareMesh(nx, ny, quadrilateral=False)  # Square mesh generated via firedrake
+
+time = 0.0  # Initial time
+timestep = 0  # Initial timestep
+timesteps_per_adapt = 10
```

Comment:
`timesteps_per_adapt = 10  # Number of simulation time steps between each instance of mesh adaptation`